src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -505,7 +505,12 @@ internal void ReportDiagnostics<T>(
                         // but no argument was supplied for it then the first such method is 
                         // the best bad method.
                         case MemberResolutionKind.RequiredParameterMissing:
-                            if ((binder.Flags & BinderFlags.CollectionExpressionConversionValidation) != 0)
+                            // Special case for collection expressions and 'params' arrays. Note: if the collection
+                            // expression has a 'with' element, we want to so normal diagnostic reporting as we want
+                            // to give accurate information about the arguments they supplied and the end construct
+                            // signature they were trying to create.
+                            if ((binder.Flags & BinderFlags.CollectionExpressionConversionValidation) != 0 &&
+                                !binder.BindingCollectionExpressionWithArguments)
```

Comment:
this is the crux of hte change.  We previously had this bit here which caused the binder to issue a different set of diagnostic messages for collection exprs.  That fine for most cases.  However, if we're actually binding the with-element, we don't want that.  So this binder flag helps allow the caller to opt in/out of this behavior.

Reply:
I understand that `params` scenario is different, since `with` cannot be used. But for a collection expression without `with`, why aren't we reporting the more informative diagnostic? This would tell the user to add a `with` for missing arguments

Reply:
Not blocking since relates to diagnostic quality, but I'd prefer we always present the more informative diagnostic when it makes sense and can help users. I'm not sure what makes you say this is a niche case

Reply:
Trivial examination of the ecosystem, collections, apis, etc.  There just aren't that many cases of collections that require arguments.  Even having that be something we supported was more something we just threw in primarily because we didn't see a problem with it, not because we actually viewed it as something important. 

(i'm not sure if we even have an ask for this scenario.  it's more that it was something we figured was non problematic to do).  So deep diagnostic quality there is genuinely just not something i'm particularly striving for :)